File: app/src/main/java/com/lazysoul/kotlinwithandroid/common/RxPresenter.java
package com.lazysoul.kotlinwithandroid.common;

import io.reactivex.disposables.CompositeDisposable;
import io.reactivex.disposables.Disposable;

/**
 * Created by Lazysoul on 2017. 7. 15..
 */

public class RxPresenter {

    private CompositeDisposable disposables;

    protected RxPresenter() {
        disposables = new CompositeDisposable();
    }

    protected void dispose() {
        if (null != disposables && !disposables.isDisposed()) {
            disposables.dispose();
        }
    }

    public void add(Disposable disposable) {
        disposables.add(disposable);
    }
}

File: app/src/main/kotlin/com/lazysoul/kotlinwithandroid/common/RxPresenter.kt
package com.lazysoul.kotlinwithandroid.common

import io.reactivex.disposables.CompositeDisposable
import io.reactivex.disposables.Disposable

/**
 * Created by Lazysoul on 2017. 7. 15..
 */

open class RxPresenter protected constructor() {

    private val disposables: CompositeDisposable by lazy { CompositeDisposable() }

    protected fun dispose() {
        if (!disposables.isDisposed) {
            disposables.dispose()
        }
    }

    fun add(disposable: Disposable) {
        disposables.add(disposable)
    }
}
